### PR TITLE
CNS-632: CU tracking - part 1

### PR DESCRIPTION
**Pull request by oren-lava**
In this PR I've only implemented the CU tracker. Note that in its current form, it's only counting the CU without resetting them. This task is continued on CNS-633: CU tracking - part 2

### CODE REVIEW

File: x/conflict/types/query.pb.go
Comment (Yaroms) [inline x/conflict/types/query.pb.go:1]: undo this change?

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 1]
```diff
@@ -231,6 +233,32 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 			details["rewardedCU"] = strconv.FormatInt(reward.Quo(coinsPerCu).TruncateInt64(), 10)
 		}
 
+		// track the provider's CU (after QoS influence)
```
Comment (Yaroms) [inline x/pairing/keeper/msg_server_relay_payment.go:236]: can we add this to the chargeCU method?
we already fetch the project and subscription there

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 2]
```diff
@@ -231,6 +233,32 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
...
+				utils.Attribute{Key: "sub_consumer", Value: proj.Subscription},
+			)
+		}
+		cuAfterQos := uint64(reward.Quo(coinsPerCu).TruncateInt64())
+		err = k.subscriptionKeeper.AddTrackedCu(ctx, sub.Consumer, relay.Provider, relay.SpecId, cuAfterQos, sub.Block)
+		if err != nil {
+			return nil, err
+		}
+
+		utils.LogLavaEvent(ctx, logger, subscriptiontypes.AddTrackedCuEventName, map[string]string{
```
Comment (Yaroms) [inline x/pairing/keeper/msg_server_relay_payment.go:256]: lets use the already exists event here, it might already contain the relevant fields

File: x/conflict/types/query.pb.go
Comment (oren-lava) [reply x/conflict/types/query.pb.go:1]: done. see https://github.com/lavanet/lava/commit/ddf3113fc8632e5dea14e5825f1f26c914758a13

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 1 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/ddf3113fc8632e5dea14e5825f1f26c914758a13

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 2 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/ddf3113fc8632e5dea14e5825f1f26c914758a13

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 3]
```diff
@@ -289,16 +261,19 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
...
-		utils.LogLavaEvent(ctx, logger, types.RelayPaymentEventName, successDetails, "New Proof Of Work Was Accepted")
 
-		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay)
+		cuAfterQos := uint64(reward.Quo(coinsPerCu).TruncateInt64())
+		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay, cuAfterQos)
 		if err != nil {
 			return nil, utils.LavaFormatError("Failed charging CU to project and subscription", err)
 		}
 
+		details["tracked_cu_added"] = strconv.FormatUint(cuAfterQos, 10)
```
Comment (Yaroms) [inline x/pairing/keeper/msg_server_relay_payment.go:271]: use the existing "rewardedCU" instead of tracked_cu_added

Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/723e53e095475d3d350262fc2e363f81f51dd938

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 4]
```diff
@@ -261,16 +261,19 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
...
 
 		details["relayNumber"] = strconv.FormatUint(relay.RelayNum, 10)
-		// differentiate between different relays by providing the index in the keys
-		successDetails := appendRelayPaymentDetailsToEvent(details, uint64(relayIdx))
-		// calling the same event repeatedly within a transaction just appends the new keys to the event
-		utils.LogLavaEvent(ctx, logger, types.RelayPaymentEventName, successDetails, "New Proof Of Work Was Accepted")
 
-		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay)
+		cuAfterQos := uint64(reward.Quo(coinsPerCu).TruncateInt64())
+		err = k.chargeComputeUnitsToProjectAndSubscription(ctx, clientAddr, relay, cuAfterQos)
```
Comment (omerlavanet) [inline x/pairing/keeper/msg_server_relay_payment.go:266]: the name of the function is misleading since it's also crediting CU to the provider not only charging

**Review by omerlavanet — DISMISSED** [review]

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 4 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go]: done. see https://github.com/lavanet/lava/commit/1916d2933243bfea20dc6e1f70af7ca822a907b2

**Review by Yaroms — APPROVED** [review]

**Review by Yaroms — APPROVED** [review]
